goalie/function_data.py

```diff
@@ -16,7 +14,7 @@
 ]
 
 
-class FunctionData(abc.ABC):
+class FunctionData:
```

Comment:
I'd like to keep this class as an ABC. I now realise I should've given its `__init__` method an `abc.abstractmethod` property. Please add this and the lint check will be okay.

Reply:
Thanks, makes sense! I wasn't sure what your original intention was :) done in 9dbf4b7dc8570d891dcd3ca387d5999d14b3b8c0